lib/stack_master/sparkle_formation/user_data_file.rb

```diff
@@ -0,0 +1,71 @@
+require 'sparkle_formation'
+require 'erubis'
+
+class SparkleFormation
+  module SparkleAttribute
+    module Aws
+      class SfEruby < Erubis::Eruby
+        include Erubis::ArrayEnhancer
+
+        def add_expr(src, code, indicator)
+          case indicator
+            when '='
+              src << " #{@bufvar} << (" << code << ');'
+            else
+              super
+          end
+        end
+      end
+
+      class TemplateContext < AttributeStruct
```

Comment:
Why would that be better? Variables require defining methods, so this throw away class means no side effects between multiple user data evaluations in the same template/process, if we patched SF then we'd need to clean those up so I prefer this approach instead.